src/Illuminate/Database/Schema/Blueprint.php

```diff
@@ -450,6 +451,10 @@ public function char($column, $length = 255)
      */
     public function string($column, $length = 255)
     {
+        if ($this->charset == 'utf8mb4' && $length == 255) {
```

Comment:
@GrahamCampbell Hi, i add the check and the charset parameter of Blueprint's construct. Is that right?